### PR TITLE
Gamestore fix

### PR DESCRIPTION
**Pull request by Geekbuddy**
Fixed Instant Rewards Access, Temple Teleport, Prey Wildcards, Charms offer, Jousting Eagle Baby id.

### CODE REVIEW

File: data/modules/scripts/gamestore/gamestore.lua [hunk 1]
```diff
@@ -4171,12 +4171,22 @@ GameStore.Categories = {
...
-          type = GameStore.OfferTypes.OFFER_TYPE_ITEM,
+          type = GameStore.OfferTypes.OFFER_TYPE_INSTANT_REWARD_ACCESS,
         },
+        {                
+          icons = { "Charm_Expansion_Offer.png" },
+          name = "Charm Expansions",
+          price = 450,
+          id = 65011,                
+          count = 250,
+                    description = "<i>No matter where you are in Tibia, claim your Charms on the spot!</i>\n\n{character}\n{info} added to your charms.",
```
Comment (dudantas) [inline data/modules/scripts/gamestore/gamestore.lua]: You can indent this?

**Comment by majestyotbr** [comment]
Can you fix Temple Teleport and Jousting Eagle Baby too? Their image does not appear (store repo contain the images).

**Comment by Geekbuddy** [comment]
> Can you fix Temple Teleport and Jousting Eagle Baby too? Their image does not appear (store repo contain the images).

Yes i will fix this now aswell.

File: data/modules/scripts/gamestore/gamestore.lua (hunk 1 above)
Comment (Geekbuddy) [reply data/modules/scripts/gamestore/gamestore.lua]: For now its just adds charms to charms amount so you can unlock with those, but i dont think thats is the way it works. But untill we found a way to fix that.

**Comment by majestyotbr** [comment]
Can you revert Hireling Cook and Cook Dress icon name?
Because the icon name is: https://github.com/opentibiabr/store/blob/master/64/Hireling_Cook.png

**Comment by Geekbuddy** [comment]
> Can you revert Hireling Cook icon name?
> Because the icon name is: https://github.com/opentibiabr/store/blob/master/64/Hireling_Cook.png

Yes i will fix that. i just update store images from our github. 👍

**Comment by Geekbuddy** [comment]
> Can you revert Hireling Cook and Cook Dress icon name?
> Because the icon name is: https://github.com/opentibiabr/store/blob/master/64/Hireling_Cook.png

Jousting_Eagle_Baby.png seems to be broken. Becouse has everything right enterd.

**Comment by majestyotbr** [comment]
> Jousting_Eagle_Baby.png seems to be broken. Becouse has everything right entered.

I looked in items.xml and the correct id is 36297.

File: data/modules/scripts/gamestore/gamestore.lua (hunk 1 above)
Comment (majestyotbr) [inline data/modules/scripts/gamestore/gamestore.lua]: Can you indent this?

File: data/modules/scripts/gamestore/gamestore.lua [hunk 2]
```diff
@@ -3815,7 +3815,7 @@ GameStore.Categories = {
           type = GameStore.OfferTypes.OFFER_TYPE_HIRELING_SEXCHANGE,
         },
         {
-          icons = { "Hireling_Cook.png" },
+          icons = { "Hireling_Cooking.png" },
```
Comment (majestyotbr) [inline data/modules/scripts/gamestore/gamestore.lua]: `icons = { "Hireling_Cook.png" },`

File: data/modules/scripts/gamestore/gamestore.lua [hunk 3]
```diff
@@ -3884,7 +3884,7 @@ GameStore.Categories = {
           type = GameStore.OfferTypes.OFFER_TYPE_HIRELING_OUTFIT,
         },
         {
-          icons = { "Hireling_Cook.png" },
+          icons = { "Hireling_Cooking.png" },
```
Comment (majestyotbr) [inline data/modules/scripts/gamestore/gamestore.lua]: `icons = { "Hireling_Cook.png" },`

File: data/modules/scripts/gamestore/init.lua [hunk 4]
```diff
@@ -531,16 +532,19 @@ function Player.canBuyOffer(self, offer)
...
 			end
 		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_EXPBOOST then
-			if (self:getStorageValue(51052) == 6 and (os.time() - self:getStorageValue(51053)) < 86400) then
+			if self:getStorageValue(51052) == 6 then
 				disabled = 1
 				disabledReason = "You can't buy XP Boost for today."
 			end
+			if (os.time() - self:getStorageValue(51053) < 86400) then
+				disabled = 1
+				disabledReason = "You allready have an active XP boost."
```
Comment (majestyotbr) [inline data/modules/scripts/gamestore/init.lua]: `disabledReason = "You already have an active XP boost."`

**Comment by majestyotbr** [comment]
Why did you removed the offer about charms of the gamestore.lua? It's not working? 
If not, you will need to revert the changes on init.lua about the charms.

**Comment by Geekbuddy** [comment]
> Why did you removed the offer about charms of the gamestore.lua? It's not working?
> If not, you will need to revert the changes on init.lua about the charms.

It works but dunno what to write in description and how mutch of the charms they get. Can´t find it anywhere.

**Comment by majestyotbr** [comment]
```
        {                
          icons = { "Charm_Expansion_Offer.png" },
          name = "Charm Expansion",
          price = 450,
          id = 65011, 
          description = "<i>Assign as many of your unlocked Charms as you like and get a 25% discount whenever you are removing a Charm from a creature!</i>\n\n{character}\n{once}",
          type = GameStore.OfferTypes.OFFER_TYPE_CHARMS,
        },
```

**Comment by Geekbuddy** [comment]
> ```
>         {                
>           icons = { "Charm_Expansion_Offer.png" },
>           name = "Charm Expansion",
>           price = 450,
>           id = 65011, 
>           description = "<i>Assign as many of your unlocked Charms as you like and get a 25% discount whenever you are removing a Charm from a creature!</i>\n\n{character}\n{once}",
>           type = GameStore.OfferTypes.OFFER_TYPE_CHARMS,
>         },
> ```

Ok its done.

File: data/modules/scripts/gamestore/gamestore.lua [hunk 5]
```diff
@@ -4171,12 +4171,22 @@ GameStore.Categories = {
...
+          price = 100,              
+          id = 65004,
           count = 30,
           description = "<i>No matter where you are in Tibia, claim your daily reward on the spot!</i>\n\n{character}\n{info} added to your reward wall\n{info} maximum amount that can be owned by character: 90",
-          type = GameStore.OfferTypes.OFFER_TYPE_ITEM,
+          type = GameStore.OfferTypes.OFFER_TYPE_INSTANT_REWARD_ACCESS,
         },
+        {                
+          icons = { "Charm_Expansion_Offer.png" },
+          name = "Charm Expansions",
```
Comment (majestyotbr) [inline data/modules/scripts/gamestore/gamestore.lua]: `name = "Charm Expansion",`

File: data/modules/scripts/gamestore/gamestore.lua [hunk 6]
```diff
@@ -4171,12 +4171,22 @@ GameStore.Categories = {
...
+          price = 450,
+          id = 65005,                
+          count = 250,
```
Comment (MajestyOTBR) [inline data/modules/scripts/gamestore/gamestore.lua:4186]: In global, does not have `count = 250,`

**Comment by MajestyOTBR** [comment]
When i try to buy Charm Expansion:
```
Gamestore: Purchase failed due to an unhandled script error.
        Stacktrace: data/modules/scripts/gamestore/init.lua:1179: attempt to perform arithmetic on local 'offerCount' (a nil value)
```

**Comment by MajestyOTBR** [comment]
Weird, now it worked...

File: data/modules/scripts/gamestore/gamestore.lua (hunk 6 above)
Comment (MajestyOTBR) [inline data/modules/scripts/gamestore/gamestore.lua:4186]: `count = 1,` ?

Comment (majestyotbr) [inline data/modules/scripts/gamestore/gamestore.lua]: `count = 1,`

**Comment by MajestyOTBR** [comment]
Now, i think it's correct.

**Comment by MajestyOTBR** [comment]
Now, i think it's correct. Just do the last changes i suggested.

**Review by majestyotbr — APPROVED** [review]

**Pull request merged**